pkgs/tools/networking/rosenpass/default.nix

```diff
@@ -1,79 +1,46 @@
 { lib
-, targetPlatform
 , fetchFromGitHub
+, nixosTests
 , rustPlatform
+, targetPlatform
 , cmake
-, makeWrapper
-, pkg-config
-, removeReferencesTo
-, coreutils
-, findutils
-, gawk
-, wireguard-tools
-, bash
 , libsodium
+, pkg-config
 }:
-
-let
-  rpBinPath = lib.makeBinPath [
-    coreutils
-    findutils
-    gawk
-    wireguard-tools
-  ];
-in
 rustPlatform.buildRustPackage rec {
   pname = "rosenpass";
   version = "0.2.0";
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-r7/3C5DzXP+9w4rp9XwbP+/NK1axIP6s3Iiio1xRMbk=";
+    hash = "sha256-r7/3C5DzXP+9w4rp9XwbP+/NK1axIP6s3Iiio1xRMbk=";
   };
 
   cargoHash = "sha256-g2w3lZXQ3Kg3ydKdFs8P2lOPfIkfTbAF0MhxsJoX/E4=";
 
   nativeBuildInputs = [
     cmake # for oqs build in the oqs-sys crate
-    makeWrapper # for the rp shellscript
     pkg-config # let libsodium-sys-stable find libsodium
-    removeReferencesTo
     rustPlatform.bindgenHook # for C-bindings in the crypto libs
   ];
 
-  buildInputs = [
-    bash # for patchShebangs to find it
-    libsodium
-  ];
-
-  # otherwise pkg-config tries to link non-existent dynamic libs during the build of liboqs
-  PKG_CONFIG_ALL_STATIC = true;
+  buildInputs = [ libsodium ];
 
-  # liboqs requires quite a lot of stack memory, thus we adjust the default stack size picked for
-  # new threads (which is used by `cargo test`) to be _big enough_
-  RUST_MIN_STACK = 8 * 1024 * 1024; # 8 MiB
+  # liboqs requires quite a lot of stack memory, thus we adjust
+  # Increase the default stack size picked for new threads (which is used
+  # by `cargo test`) to be _big enough_.
+  # Only set this value for the check phase (not as an environment variable for the derivation),
+  # because it is only required in this phase.
+  preCheck = "export RUST_MIN_STACK=${builtins.toString (8 * 1024 * 1024)}"; # 8 MiB
```

Comment:
```suggestion
  # liboqs requires quite a lot of stack memory, thus we adjust
  # the default stack size picked for new threads (which is used
  # by `cargo test` in checkPhase only) to be _big enough_.
  preCheck = ''
    export RUST_MIN_STACK=${toString (8 * 1024 * 1024)}
  '';
```

Reply:
For which reason did you change this from `"` to `''`.

Reply:
Wasn't it like that only the indented string is concatenated when merging attribute sets, while the single line string literal is overwritten?

Reply:
Nix doesn't know the difference AFAIK, it's all just strings.
Using multiline strings right away will minimize the diff when expanding the script, so I see this as a stylistic suggestion one can take or ignore.

Reply:
This was resolved by bumping Rosenpass to a revision that does not need the environment variable to be set anymore.